test/test-lib/fixtures.ts

```diff
@@ -471,6 +471,60 @@ const loaders: types.Dictionary<LoaderFunc> = {
 		});
 	},
 
+	device_environment_variables: async (jsonData, fixtures) => {
+		const user = await fixtures.users[jsonData.user];
+		if (user == null) {
+			logErrorAndThrow(`Could not find user: ${jsonData.user}`);
+		}
+		const device = await fixtures.devices[jsonData.device];
+		if (device == null) {
+			logErrorAndThrow(`Could not find device: ${jsonData.device}`);
+		}
+
+		return await requestResource({
+			resource: 'device_environment_variable',
+			body: {
+				device: device.id,
+				name: jsonData.name,
+				value: jsonData.value,
+			},
+			user,
+		});
+	},
+	device_service_environment_variables: async (jsonData, fixtures) => {
+		const user = await fixtures.users[jsonData.user];
+		if (user == null) {
+			logErrorAndThrow(`Could not find user: ${jsonData.user}`);
+		}
+
+		const device = await fixtures.devices[jsonData.device];
+		if (device == null) {
+			logErrorAndThrow(`Could not find device: ${jsonData.device}`);
+		}
+
+		const service = await fixtures.services[jsonData.service];
+		if (service == null) {
+			logErrorAndThrow(`Could not find service: ${jsonData.service}`);
+		}
+
+		const {
+			d: [si],
+		} = await requestResource({
+			resource: `service_install(device=${device.id},installs__service=${service.id})`,
+			method: 'GET',
+			user,
+		});
```

Comment:
It makes more sense imo to just fetch this with pinejs-client, eg
```suggestion
		const si = await api.resin.get({
			resource: 'service_install'
			passthrough: { req: permissions.root },
			id: {
				device: device.id,
				installs__service: service.id,
			},
		});
```
rather than the weird resource with embedded params.